cli/module_loader.rs

```diff
@@ -113,44 +112,34 @@ impl ModuleLoader for CliModuleLoader {
     &self,
     op_state: Rc<RefCell<OpState>>,
     _load_id: ModuleLoadId,
-    module_specifier: &ModuleSpecifier,
-    maybe_referrer: Option<String>,
-    is_dyn_import: bool,
+    specifier: &ModuleSpecifier,
+    _maybe_referrer: Option<String>,
+    is_dynamic: bool,
   ) -> Pin<Box<dyn Future<Output = Result<(), AnyError>>>> {
-    let module_specifier = module_specifier.clone();
+    let specifier = specifier.clone();
     let target_lib = self.target_lib.clone();
     let maybe_import_map = self.import_map.clone();
     let state = op_state.borrow();
 
     // Only "main" module is loaded without permission check,
     // ie. module that is associated with "is_main" state
     // and is not a dynamic import.
-    let permissions = if self.is_main && !is_dyn_import {
+    let permissions = if self.is_main && !is_dynamic {
       Permissions::allow_all()
     } else {
       state.borrow::<Permissions>().clone()
     };
     let program_state = state.borrow::<Arc<ProgramState>>().clone();
     drop(state);
 
-    // TODO(bartlomieju): I'm not sure if it's correct to ignore
-    // bad referrer - this is the case for `Deno.core.evalContext()` where
-    // `ref_str` is `<unknown>`.
-    let maybe_referrer = if let Some(ref_str) = maybe_referrer {
-      ModuleSpecifier::resolve_url(&ref_str).ok()
-    } else {
-      None
-    };
-
     // TODO(bartlomieju): `prepare_module_load` should take `load_id` param
     async move {
       program_state
         .prepare_module_load(
-          module_specifier,
-          maybe_referrer,
```

Comment:
So in case of dynamic import `maybe_referrer` will have value - if trying to load `module_specifier` results in eg. `404` we can display the error with information that this failed import originated from `maybe_referrer`.

Reply:
But then the module graph would have to have the concept of referrer on add... which doesn't make sense... As mentioned before, the module graph silently squashes these error and allows them to be done on the `fetch_compiled_module` and why `error_014_catch_dynamic_import_error` passes (which includes the referrer).